### PR TITLE
Create Benchmark based on Rat's Life

### PR DESCRIPTION
**Pull request by lukicdarkoo**
**Description**

There are three main objectives:
1. The old Navigation2 configuration is not compatible with the new Foxy release, it has to be fixed.
2. The benchmark has to quantify the difference between the real and simulated robots in a few scenarios.
3. Update relevant worlds and protos to work with Webots 2020b.

**Affected Packages**
  - webots_ros2_core
  - webots_ros2_examples
  - webots_ros2_epuck

**Tasks**
  - [x] Update configuration for Navigation2 to match Foxy
  - [x] Fix world coordinate system to work in Webots 2020b (not all, only `epuck_world.wbt`, `khepera4_example.wbt`, `turtlebot3_burger_example.wbt`, `tiago++_example.wbt`) 
  - [x] Create new world `rats_life_benchmark.wbt`
  - [x] Fix protos to work in Webots 2020b 
    - [x] IMU in TurtleBot3 Burger stopped working as the parent node didn't have physics declared
    - [x] Controller arguments
  - [x] Add Khepera IV driver to support LaserScan from the distance sensors
  - [x] Add ROS maps for `epuck_world.wbt` and `rats_life_benchmark.wbt` so the Navigation2 package can use them to better navigation
    - [x] Use particle filter in the navigation examples

---

  - [x] Master thesis specific
    - [x] Benchmark
      - [x] Mapping, real vs simulated (3 experiments each + ground truth, find IoU for each, express difference in mean IoU)
        - [x] Create ground truth map
      - [x] Navigation, real vs simulated (3 experiments each, find IoU for each, express difference in mean IoU)
      - [x] Mapping, hardware abstraction
      - [x] Navigation, hardware abstraction
    - [x] Use the opportunity to make videos of the real and simulated robots

### CODE REVIEW

File: webots_ros2_examples/worlds/turtlebot3_burger_example.wbt [hunk 1]
```diff
@@ -109,12 +116,12 @@ Wall {
   }
 }
 Radiator {
-  translation 7.5 0.33 -4.87
-  rotation 0 1 0 -1.5707996938995747
+  translation 7.5 4.87 0.33
+  rotation -0.7070262765728255 0.707187276635805 5.0851819892091665e-09 -3.1412653071795864
```
Comment (DavidMansolino) [inline webots_ros2_examples/worlds/turtlebot3_burger_example.wbt:120]:
```suggestion
  translation 7.33 4.87 0.38
  rotation -0.5771735891339322 0.5774385889452899 0.5774385889452899 -2.094255307179586
```

File: webots_ros2_epuck/protos/LegoTallInterval.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#VRML_SIM R2020b utf8
```
[the remark this answers is not in the document]
Comment (lukicdarkoo) [reply webots_ros2_epuck/protos/LegoTallInterval.proto:1]: It comes from:
https://github.com/cyberbotics/webots/blob/master/projects/samples/contests/ratslife/protos/LegoInterval.proto

But I needed to increase height of it, so I simplified it and changed its height.

**Comment by lukicdarkoo** [comment]
> The `universal_robot_rviz.wbt` world needs either a header to R2020a either a `NUE` coordinate system.

I see, but this PR supposed to add only Rat's Life world with Navigation2 integration. The PR has already grown way bigger than that, so I wanted to create a new PR that will prepare all the packages for the release. If you agree? Or you think it is fine to do everything in this PR?

In this PR only the following worlds are fixed `epuck_world.wbt`, `khepera4_example.wbt`, `turtlebot3_burger_example.wbt` and `tiago++_example.wbt`.

File: webots_ros2_epuck/launch/rats_life_waypoints_launch.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    mapper = Node(
+        package='webots_ros2_epuck',
+        node_executable='simple_mapper',
+        output='screen',
+        parameters=[{'use_sim_time': use_sim_time, 'fill_map': True}],
+    )
+
+    send_waypoints = ExecuteProcess(
+        # The goal has to be sent once the simulation is up and running, therefore we wait 10s first
+        cmd=[f'sleep 10 && ros2 action send_goal /FollowWaypoints nav2_msgs/action/FollowWaypoints \'{get_waypoints()}\''],
```
[the remark this answers is not in the document]
Comment (lukicdarkoo) [reply webots_ros2_epuck/launch/rats_life_waypoints_launch.py:142]: I know :/ I will check if I can do something about it

**Comment by DavidMansolino** [comment]
> I see, but this PR supposed to add only Rat's Life world with Navigation2 integration. The PR has already grown way bigger than that, so I wanted to create a new PR that will prepare all the packages for the release. If you agree? Or you think it is fine to do everything in this PR?

Yes sure, this was just to list the worlds that absolutely need some changes.

File: webots_ros2_epuck/launch/rats_life_waypoints_launch.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukicdarkoo) [reply webots_ros2_epuck/launch/rats_life_waypoints_launch.py]: Works!

I tried a nicer solution, to call the action once a state of the corresponding node is active, but it didn't work (the node is not ready at that point).

File: webots_ros2_epuck/launch/rats_life_launch.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    rviz = Node(
+        package='rviz2',
+        executable='rviz2',
+        name='rviz2',
+        arguments=['-d', os.path.join(get_package_share_directory('nav2_bringup'), 'rviz', 'nav2_default_view.rviz')],
+        parameters=[{'use_sim_time': use_sim_time}],
+        output='screen'
+    )
+
+    initial_position = ExecuteProcess(
```
[the remark this answers is not in the document]
Comment (lukicdarkoo) [reply webots_ros2_epuck/launch/rats_life_launch.py:70]: Added comments to explain vague node configurations. Let me know if more is needed.

[the remark this answers is not in the document]
Comment (DavidMansolino) [reply webots_ros2_epuck/launch/rats_life_launch.py:70]: Looks good!

**Comment by DavidMansolino** [comment]
:eyes:

File: webots_ros2_epuck/launch/rats_life_waypoints_launch.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        ),
+        launch_arguments=[
+            ('package', 'webots_ros2_epuck'),
+            ('executable', 'driver'),
+            ('world', os.path.join(package_dir, 'worlds', 'rats_life_benchmark.wbt')),
+        ],
+        condition=launch.conditions.IfCondition(use_sim_time)
+    )
+
+    # Launch Navigation2 without a localization and without costmaps
```
Comment (DavidMansolino) [inline webots_ros2_epuck/launch/rats_life_waypoints_launch.py:117]: It would be nice to say why we need it if localization and costmaps is disabled.

**Pull request merged**